sdk/servicebus/service-bus/samples-v1/javascript/receiveMessagesStreaming.js

```diff
@@ -33,18 +33,24 @@ async function main() {
   do {
     const receiver = queueClient.createReceiver(ReceiveMode.peekLock);
 
-    const receiverPromise = new Promise((resolve, _reject) => {
+    const receiverPromise = new Promise((resolve, reject) => {
       const onMessageHandler = async (brokeredMessage) => {
         console.log(`Received message: ${brokeredMessage.body}`);
         await brokeredMessage.complete();
       };
 
       const onErrorHandler = (err) => {
         if (err.retryable === true) {
+          enableReceiverRecovery = true;
           console.log("Receiver will be recreated. A recoverable error occurred:", err);
           resolve();
         } else {
+          // Set `enableReceiverRecovery` to false to break out of the loop for a non-retryable error
+          // since the error might be hinting at a deeper issue such as faulty configuration
+          // or a non-existent queue which requires attention from the user.
+          enableReceiverRecovery = false;
           console.log("Error occurred: ", err);
+          reject(err);
```

Comment:
I'm wondering if we should just remove `enableReceiverRecovery`. Throwing is enough to break out of the loop (since we're not try/catching around the `await receiverPromise`).

So perhaps it's just a 

```typescript
while (true) {

}
```

instead?

Reply:
So, just throw for all errors from onError?

Reply:
You mean
```ts

export async function main() {
  const sbClient = ServiceBusClient.createFromConnectionString(connectionString);

  // If receiving from a Subscription, use `createSubscriptionClient` instead of `createQueueClient`
  const queueClient = sbClient.createQueueClient(queueName);

  // To receive messages from sessions, use createSessionReceiver instead of createReceiver or look at
  // the sample in sessions.ts file

  do {
    const receiver = queueClient.createReceiver(ReceiveMode.peekLock);

    const receiverPromise = new Promise<void>((resolve, reject) => {
      const onMessageHandler: OnMessage = async (brokeredMessage) => {
        console.log(`Received message: ${brokeredMessage.body}`);
        await brokeredMessage.complete();
      };

      const onErrorHandler: OnError = (err) => {
        if ((err as MessagingError).retryable === true) {
          console.log("Receiver will be recreated. A recoverable error occurred:", err);
          resolve();
        } else {
          // Set `enableReceiverRecovery` to false to break out of the loop for a non-retryable error
          // since the error might be hinting at a deeper issue such as faulty configuration
          // or a non-existent queue which requires attention from the user.
          console.log("Error occurred: ", err);
          reject(err);
        }
      };

      receiver.registerMessageHandler(onMessageHandler, onErrorHandler, { autoComplete: false });
    });

    // This will only resolve if our receiver has failed in a way that is not recoverable.
    await receiverPromise;

    // the Service Bus package is intended to be resilient in the face of transitive issues, like network
    // interruptions. If there are continual restarts this might indicate a more serious issue, like a network
    // outage.
    console.log(`Closing previous receiver and recreating - a fatal error has occurred.`);

    // we can close the old receiver and just let the loop start again.
    await receiver.close();
  } while (true);

  await queueClient.close();
  await sbClient.close();
}
```

Reply:
Yes!

Reply:
Updated!